quickwit/quickwit-control-plane/src/control_plane.rs

```diff
@@ -273,42 +334,27 @@ impl Handler<DeleteSourceRequest> for ControlPlane {
             index_uid,
             source_id: request.source_id,
         };
-        // TODO: Notify indexing controller.
+
+        // the metastore. We should update the state of the control plane.
+        self.indexing_scheduler.on_index_change().await?;
         let response = EmptyResponse {};
         Ok(Ok(response))
     }
 }
 
-#[async_trait]
-impl Handler<NotifyIndexChangeRequest> for ControlPlane {
-    type Reply = ControlPlaneResult<NotifyIndexChangeResponse>;
-
-    async fn handle(
-        &mut self,
-        request: NotifyIndexChangeRequest,
-        _: &ActorContext<Self>,
-    ) -> Result<Self::Reply, ActorExitStatus> {
-        debug!("Index change notification: schedule indexing plan.");
-        // TODO: Switch from async (`send_message`) to sync (`ask_for_res`).
-        self.indexing_scheduler_mailbox
-            .send_message(request)
-            .await
-            .context("error sending index change notification to index scheduler")?;
-        Ok(Ok(NotifyIndexChangeResponse {}))
-    }
-}
-
 #[async_trait]
 impl Handler<GetOpenShardsRequest> for ControlPlane {
     type Reply = ControlPlaneResult<GetOpenShardsResponse>;
 
     async fn handle(
         &mut self,
         request: GetOpenShardsRequest,
-        _: &ActorContext<Self>,
+        ctx: &ActorContext<Self>,
     ) -> Result<Self::Reply, ActorExitStatus> {
-        let response = handle_ask_res!(self.ingest_controller_mailbox.ask_for_res(request).await);
-        Ok(Ok(response))
+        Ok(self
+            .ingest_controller
+            .get_open_shards(request, ctx.progress())
```

Comment:
I've renamed this event `GetOrCreateOpenShardsRequest` to make it clear that it may create new shards. When it does, the indexing scheduler should be notified.